pkg/data/data.go

```diff
@@ -75,6 +77,21 @@ func (job *Job) Save(db *sqlx.DB) (sql.Result, error) {
 	)
 }
 
+func (job *Job) AuthSignature(secret string) string {
```

Comment:
@kevinfalting Can you help me understand this one. Golang is not the language I use at my day job so I've forgotten some details. Does referencing the value create a copy of the `Job` thus kinda being "read only" in a way?

Reply:
Yeah, totally! You're exactly right, you can either have a value receiver
```go
func (job Job)
```
or a pointer, or reference, receiver.
```go
func (job *Job)
```

Passing by value, creates a copy, any changes to the job struct will not be reflected in the caller's scope.

- https://go.dev/doc/effective_go#pointers_vs_values
- https://gobyexample.com/methods

However, I think I may want to say keep it as a pointer receiver. While it would be correct to make it a value receiver, if we want to match the other methods, they're all pointers. Sometimes it's just best to match the code around it. I'll leave this one to you. 👍

Reply:
To further drive home that I was wrong: https://dave.cheney.net/2016/03/19/should-methods-be-declared-on-t-or-t

> In short, I think that you should prefer declaring methods on *T unless you have a strong reason to do otherwise.

and then footnote 5 is thought-worthy:

> If the method does not mutate its receiver, does it need to be a method?

Reply:
Hmm, thanks for linking that article. That's super helpful! My thoughts on the footnote is that I personally prefer keeping it as a method because it keeps the code associated with the struct and is discoverable through code completion.

Reply:
Fair, I think that makes sense here. 👍